code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1255,16 +1255,55 @@
 	else
 		germ_level += n
 
+/mob/living/carbon/human/proc/check_and_regenerate_organs(var/mob/living/carbon/human/H) //Regenerates missing limbs/organs.
+	var/datum/species/S = H.species
+	var/list/species_organs = S.has_organ
+	var/list/species_limbs = S.has_limbs
+	var/list/types_of_int_organs = list()
+	for(var/obj/item/organ/internal/I in H.internal_organs)
+		types_of_int_organs |= I.type
+
+	//Removing stumps.
+	for(var/obj/item/organ/organ in H.contents)
+		if(istype(organ, /obj/item/organ/external/stump)) //Get rid of all stumps.
+			qdel(organ)
+			H.contents -= organ //Making sure the list entry is removed.
+	for(var/obj/item/organ/organ in H.organs)
+		if(istype(organ, /obj/item/organ/external/stump))
+			qdel(organ)
+			H.organs -= organ //Making sure the list entry is removed.
+	for(var/organ_name in H.organs_by_name)
+		var/obj/item/organ/organ = H.organs_by_name[organ_name]
+		if(istype(organ, /obj/item/organ/external/stump) || !organ) //The !organ is to account for mechanical limb losses, since those are handled in a way that creates indexed but null list entries instead of stumps.
+			qdel(organ)
+			H.organs_by_name -= organ_name //Making sure the list entry is removed.
+
+	//Replacing lost limbs with the species default.
+	for(var/limb_type in species_limbs)
+		if(!(limb_type in H.organs_by_name))
+			var/list/organ_data = species_limbs[limb_type]
+			var/limb_path = organ_data["path"]
+			var/obj/item/organ/external/O = new limb_path(H)
+			O.owner = H
+			H.organs |= H.organs_by_name[O.limb_name]
+
+	//Replacing lost organs with the species default.
+	for(var/index in species_organs)
+		var/organ = species_organs[index]
+		if(!(organ in types_of_int_organs))
+			var/obj/item/organ/internal/I = new organ(H)
+			I.insert(H)
+
 /mob/living/carbon/human/revive()
 
 	if(species && !(species.flags & NO_BLOOD))
 		var/blood_reagent = get_blood_name()
 		vessel.add_reagent(blood_reagent, max_blood-vessel.total_volume)
 		fixblood()
 
-	// Fix up all organs.
-	// This will ignore any prosthetics in the prefs currently.
-	species.create_organs(src)
+	//Fix up all organs and replace lost ones.
+	restore_all_organs()
+	check_and_regenerate_organs(src)

```

Comment:
Replaces missing limbs and organs via comparisons with species-standard limbs+organs and the ones the person getting rejuvenated has.